controllers/preflightvalidation_controller.go

```diff
@@ -0,0 +1,239 @@
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	kmmv1beta1 "github.com/qbarrand/oot-operator/api/v1beta1"
+	"github.com/qbarrand/oot-operator/internal/filter"
+	"github.com/qbarrand/oot-operator/internal/preflight"
+	"github.com/qbarrand/oot-operator/internal/statusupdater"
+	"github.com/qbarrand/oot-operator/internal/utils"
+)
+
+const reconcileRequeueInSeconds = 60
+
+// ClusterPreflightReconciler reconciles a PreflightValidation object
+type PreflightValidationReconciler struct {
+	client        client.Client
+	filter        *filter.Filter
+	statusUpdater statusupdater.PreflightStatusUpdater
+	preflight     preflight.PreflightAPI
+}
+
+func NewPreflightReconciler(
+	client client.Client,
+	filter *filter.Filter,
+	statusUpdater statusupdater.PreflightStatusUpdater,
+	preflight preflight.PreflightAPI) *PreflightValidationReconciler {
+	return &PreflightValidationReconciler{
+		client:        client,
+		filter:        filter,
+		statusUpdater: statusUpdater,
+		preflight:     preflight}
+}
+
+func (r *PreflightValidationReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("preflightvalidation").
+		For(&kmmv1beta1.PreflightValidation{}).
+		Watches(
+			&source.Kind{Type: &kmmv1beta1.Module{}},
+			handler.EnqueueRequestsFromMapFunc(r.filter.FindPreflightsForModule),
+			builder.WithPredicates(r.filter.PreflightReconcilerModulePredicate()),
+		).
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: 1,
+		}).
+		Complete(r)
+}
+
+//+kubebuilder:rbac:groups=kmm.sigs.k8s.io,resources=modules,verbs=get;list;watch
+//+kubebuilder:rbac:groups=kmm.sigs.k8s.io,resources=preflightvalidations,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=kmm.sigs.k8s.io,resources=preflightvalidations/status,verbs=get;update;patch
+
+// Reconcile Reconiliation entry point
+func (r *PreflightValidationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+
+	reconRes := ctrl.Result{}
+
+	log := ctrl.LoggerFrom(ctx)
+	log.Info("Start PreflightValidation Reconciliation")
+
+	pv := kmmv1beta1.PreflightValidation{}
+	err := r.client.Get(ctx, types.NamespacedName{Namespace: req.Namespace, Name: req.Name}, &pv)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("PreflightValidation reconcile success. Reconciliation object not found, probably deleted. Not reconciling")
+			return reconRes, nil
+		} else {
+			log.Error(err, "preflight validation reconcile failed to find object")
+			return reconRes, err
+		}
+	}
+
+	reconCompleted, err := r.runPreflightValidation(ctx, &pv)
+	if err != nil {
+		log.Error(err, "runPreflightValidation failed")
+		return reconRes, err
```

Comment:
```suggestion
			return ctrl.Result{}, err
```
It is the same but clearer IMO. You also do it later on this file anyway.

Reply:
fixed